extensions/wikia/DesignSystem/ds.built.js

```diff
@@ -0,0 +1,768 @@
+			}
+			var lastStyleElementInsertedAtTop = styleElementsInsertedAtTop[styleElementsInsertedAtTop.length - 1];
+			if (options.insertAt === "top") {
+				if(!lastStyleElementInsertedAtTop) {
+					styleTarget.insertBefore(styleElement, styleTarget.firstChild);
+				} else if(lastStyleElementInsertedAtTop.nextSibling) {
+					styleTarget.insertBefore(styleElement, lastStyleElementInsertedAtTop.nextSibling);
+				} else {
+					styleTarget.appendChild(styleElement);
+				}
+				styleElementsInsertedAtTop.push(styleElement);
+			} else if (options.insertAt === "bottom") {
+				styleTarget.appendChild(styleElement);
+			} else {
+				throw new Error("Invalid value for parameter 'insertAt'. Must be 'top' or 'bottom'.");
+			}
+		}
+
+		function removeStyleElement(styleElement) {
+			styleElement.parentNode.removeChild(styleElement);
+			var idx = styleElementsInsertedAtTop.indexOf(styleElement);
+			if(idx >= 0) {
+				styleElementsInsertedAtTop.splice(idx, 1);
+			}
+		}
+
+		function createStyleElement(options) {
+			var styleElement = document.createElement("style");
+			options.attrs.type = "text/css";
+
+			attachTagAttrs(styleElement, options.attrs);
+			insertStyleElement(options, styleElement);
+			return styleElement;
+		}
+
+		function createLinkElement(options) {
+			var linkElement = document.createElement("link");
+			options.attrs.type = "text/css";
+			options.attrs.rel = "stylesheet";
+
+			attachTagAttrs(linkElement, options.attrs);
+			insertStyleElement(options, linkElement);
+			return linkElement;
+		}
+
+		function attachTagAttrs(element, attrs) {
+			Object.keys(attrs).forEach(function (key) {
+				element.setAttribute(key, attrs[key]);
+			});
+		}
+
+		function addStyle(obj, options) {
+			var styleElement, update, remove;
+
+			if (options.singleton) {
+				var styleIndex = singletonCounter++;
+				styleElement = singletonElement || (singletonElement = createStyleElement(options));
+				update = applyToSingletonTag.bind(null, styleElement, styleIndex, false);
+				remove = applyToSingletonTag.bind(null, styleElement, styleIndex, true);
+			} else if(obj.sourceMap &&
+			          typeof URL === "function" &&
+			          typeof URL.createObjectURL === "function" &&
+			          typeof URL.revokeObjectURL === "function" &&
+			          typeof Blob === "function" &&
+			          typeof btoa === "function") {
+				styleElement = createLinkElement(options);
+				update = updateLink.bind(null, styleElement, options);
+				remove = function() {
+					removeStyleElement(styleElement);
+					if(styleElement.href)
+						URL.revokeObjectURL(styleElement.href);
+				};
+			} else {
+				styleElement = createStyleElement(options);
+				update = applyToTag.bind(null, styleElement);
+				remove = function() {
+					removeStyleElement(styleElement);
+				};
+			}
+
+			update(obj);
+
+			return function updateStyle(newObj) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this function anonymous by removing its name: 'function() {...}'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANamedFunctionExpression)